### ISSUE TITLE
[Feature Request] Support for Rev 1 of the Japanese version

### ISSUE THREAD

**Issue by TzakShrike** [pyriell/gs2-bugfixes#9]
Presently, only Rev 0 (the original release) of the Japanese version is supported, and Rev 1 (the PlayStation The Best release) is not supported, but this isn't indicated in the documentation. It fails gracefully because the file is checked before patching.

**Comment by TzakShrike** [comment]
FYI I'm quite willing to do most of the work for this myself, just need to find the time.

**Comment by pyriell** [comment]
I was unaware the Playstation the Best version of this existed.  I'm surprised it differed from the original in any significant way.  I guess for whatever reason they typically do, at least as far as the serial number.  That might be all that needs to change.

I've never seen an image of that version online that I can recall, though.  It's probably floating around somewhere, but I'd have to find it first.

**Comment by TzakShrike** [comment]
I have both. The binaries do differ, but I haven't delved deep enough to know how much yet.

... OK i just got home and checked my notes. There's about 10Mb worth of differences in the binaries alone. Not that it matters, but the discs are also mastered differently.

Considering Rev 1 releases generally contain bugfixes, it would be interesting to see what else changed.

**Comment by TzakShrike** [comment]
OK so I had a slightly closer look. Those 10Mb of differences are split across **586** BIN files. There are also differences in SLPM_861.68 ,  and really interestingly, in YAKUSO2.SDD.

...Aww... I checked again and the updated YAKUSO2.SDD is identical to the one in the U/C release. I still wonder what changed though.

**Comment by pyriell** [comment]
Weird.  That's about half the files on the disc too.  I've heard no reports of Konami doing significant bug-squashing in a Japanese revision.  I guess they could have, and nobody might have noticed, if it sold poorly.  I think it's probably more likely they recompiled with an updated dev kit or something, though.  Given the way the game is structured, if something in the main executable moved around, that would propagate to pretty much every module with executable code in it.

**Comment by TzakShrike** [comment]
Yeah pretty much exactly half of the binary files, and basically none of anything else.
I agree that it was probably recompiled differently. I thought it might've been with an optimization flag that was missed the first time or something, but the binaries are essentially the same size, so I don't think that could be it.
Still, the change in YAKUSO2.SDD leads me to think that any bug squashing was likely between Rev 0 and the International Releases. Maybe there are more Rev 0 specific bugs that aren't accounted for?

**Comment by pyriell** [comment]
It's possible there are more bugs that are unaccounted for.  I don't think anybody would have known about the bug with the rune unite speed boost, if it hadn't been for speed-runners planning their combat rounds so minutely that one of them realized a character went earlier than he should have.  It seems almost impossible that there aren't more bugs like that lurking somewhere.

I doubt there were any fixes from revision 0 to the international release.  All the bugs we know about are present in every version, except the ones that can be accounted for by localization.  The only two that were ever fixed (that I know of) were the name import, and the music, both of which were introduced in the US localization.  The music was fixed by proper mastering in Europe, and the name import was probably just a target of opportunity during localization in Europe.  I can't recall if the character set they used in the US region was complete for Europe in Suikoden and Suikoden 2, but needing to account for differences is possibly the only reason that work got done.  I suppose a third is fixing some of the text that was untranslated in the US.

I mean, the Matilda Glitch is fixed by changing one byte of data for a map object, and it seems impossible that nobody in Japan discovered that exploit in the year or two it took for the game to come out in other regions.  That didn't get fixed until the PSP compilation was done in 2006.
